packages/search/addon/services/navi-search/navi-asset-search-provider.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright 2020, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ *
+ * This service is used to search for reports and dashboards stored in the persistence layer.
+ */
+
+import { inject as service } from '@ember/service';
+import NaviBaseSearchProviderService from '../navi-base-search-provider';
+import { keepLatestTask } from 'ember-concurrency-decorators';
+import { pluralize } from 'ember-inflector';
+
+export default class NaviAssetSearchProviderService extends NaviBaseSearchProviderService {
+  /**
+   * @property {Ember.Service} store
+   */
+  @service store;
+
+  /**
+   * @property {Ember.Service} user
+   */
+  @service user;
+
+  /**
+   * @property {String} _displayComponentName
+   * @private
+   */
+  _displayComponentName = 'navi-search-result/asset';
+
+  /**
+   * @method _parseQueryString – Parses string query to search parameters
+   * @private
+   * @param {String} query
+   * @param {String} type
+   * @returns {Object} query object
+   */
+  _parseQueryString(query, type) {
+    let searchParams;
+
+    if (typeof query == 'string' && query) {
+      if (type === 'report') {
+        searchParams = {
+          title: query,
+          request: query
+        };
+      } else if (type === 'dashboard') {
+        searchParams = {
+          title: query
+        };
+      }
+    }
+
+    return searchParams;
+  }
+
+  /**
+   * @method _constructSearchQuery – Constructs the query filter parameters adhering to the RSQL standard
+   * @private
+   * @param {Object} searchParams
```

Comment:
Can we update? Also can we add a detailed description, construction of `or` filters and `and` author filter?